llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1443,41 +1459,26 @@ bool NVPTXDAGToDAGISel::tryStoreVector(SDNode *N) {
   // - for integer type, always use 'u'
   const unsigned TotalWidth = StoreVT.getSimpleVT().getSizeInBits();
 
-  SmallVector<SDValue, 12> Ops;
-  SDValue N2;
-  unsigned VecType;
-  unsigned ToTypeWidth;
-
-  switch (N->getOpcode()) {
-  case NVPTXISD::StoreV2:
-    VecType = NVPTX::PTXLdStInstCode::V2;
-    Ops.append({N->getOperand(1), N->getOperand(2)});
-    N2 = N->getOperand(3);
-    ToTypeWidth = TotalWidth / 2;
-    break;
-  case NVPTXISD::StoreV4:
-    VecType = NVPTX::PTXLdStInstCode::V4;
-    Ops.append({N->getOperand(1), N->getOperand(2), N->getOperand(3),
-                N->getOperand(4)});
-    N2 = N->getOperand(5);
-    ToTypeWidth = TotalWidth / 4;
-    break;
-  default:
-    return false;
-  }
+  unsigned NumElts = getLoadStoreVectorNumElts(N);
+  SmallVector<SDValue, 16> Ops;
+  // Append the operands from 1 to NumElts, inclusive
+  const SDUse *FirstStoredVal = N->ops().begin() + 1;
+  Ops.append(FirstStoredVal, FirstStoredVal + NumElts);
```

Comment:
I think it could be simplified by using `ArrayRef::slice`.

E.g.  something like this:
```
SmallVector<SDValue, 16> Ops(N->ops().slice(1, NumElts));
```